terminal/bundles/org.eclipse.terminal.view.ui/src/org/eclipse/terminal/view/ui/internal/handler/LaunchTerminalCommandHandler.java

```diff
@@ -172,11 +173,11 @@ private void executeDelegate(ISelection selection, ILauncherDelegate delegate) t
 	}
 
 	private void executeDelegate(Map<String, Object> properties, ILauncherDelegate delegate) throws ExecutionException {
-		try {
-			delegate.execute(properties);
-		} catch (Exception e) {
-			throw new ExecutionException(e.getMessage(), e);
-		}
+		delegate.execute(properties).whenComplete((r, e) -> {
```

Comment:
```suggestion
		delegate.execute(properties).exceptionally(e -> {
```

then you can be sure it only triggers on error.